### PR TITLE
Refact region destinations representation

### LINKED ISSUE #35: Represent region destinations as set of vertices
Region destinations is being represented as ArrayList<String> all over the place (rbr package). Instead we should prefer representing it as Set<Vertex>. It should simplify many code segments. The main place that hold that evil kind of representation is Region.java (I recommend starting there).

### PR DESCRIPTION
**Pull request by rmilfont**
Use Set instead of ArrayList collection to represent region destination. Fix #35.

### CODE REVIEW

File: src/rbr/Region.java [hunk 1]
```diff
@@ -1,19 +1,22 @@
 package rbr;
 
 import util.Range;
+import util.Vertex;
```
Comment (jmferreiratech) [inline src/rbr/Region.java]: Use the \* here to squash the imports.

File: src/rbr/Region.java [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
+import java.util.Set;
 
 public class Region {
 	// @ToDo inputPortsSet and outputPortsSet could actually be Sets. This would ease many operations on them:
 	// @ToDo isSubset, getOpMerged (intersection), getIpMerged (union) and so on.
 	private String inputPortsSet;
 	private String outputPortsSet;
 	private Range box;
-	private ArrayList<String> destinations = new ArrayList<>();
+	private Set<Vertex> destinations = new HashSet<>();
```
Comment (jmferreiratech) [inline src/rbr/Region.java]: We don't need to allocate it as this is passed as parameter.

File: src/rbr/Region.java [hunk 3]
```diff
@@ -1,19 +1,22 @@
...
 
-	public Region(String ip, ArrayList<String> destinations, String op) {
+	public Region(String ip, Set<Vertex> destinations, String op) {
 		this.destinations = destinations;
```
Comment (jmferreiratech) [inline src/rbr/Region.java]: Maybe is more safe to use a copy constructor here. The set can be manipulated externally...

File: src/rbr/Region.java [hunk 4]
```diff
@@ -120,14 +124,12 @@ private static String getIpMerged(Region r1, Region r2) {
...
+	Set<Vertex> outsiders() {
+		Set<Vertex> result = new HashSet<>();
 		for (int x = this.box().min(0); x <= this.box().max(0); x++) {
 			for (int y = this.box().min(1); y <= this.box().max(1); y++) {
-				String name = x + "." + y;
-				if (!this.destinations().contains(name)) {
-					result.add(name);
-				}
+				Vertex vertex = new Vertex(x + "." + y);
+				result.add(vertex);
```
Comment (jmferreiratech) [inline src/rbr/Region.java]: You've missed a condition here and are adding all vertices...

**Review by jmferreiratech — CHANGES_REQUESTED** [review]

File: src/rbr/Region.java (hunk 1 above)
Comment (rmilfont) [reply src/rbr/Region.java]: Done.

File: src/rbr/Region.java (hunk 2 above)
Comment (rmilfont) [reply src/rbr/Region.java]: Done.

File: src/rbr/Region.java (hunk 4 above)
Comment (rmilfont) [reply src/rbr/Region.java]: Done.

File: src/rbr/Region.java (hunk 3 above)
Comment (rmilfont) [reply src/rbr/Region.java]: Done.

**Review by jmferreiratech — APPROVED** [review]

**Pull request merged**